README.md

```diff
@@ -1 +1,162 @@
-# node-mocha
\ No newline at end of file
+# Mocha Testing Framework
+
+## Objectives
+
+1. Describe how to install Mocha and Chai Expect
+1. Describe how to create a simple test
+1. Describe how to launch a test
+
+## Overview
+
+Imagine that you are at a hackathon, and you're pressed for time to finish your app. You make the last minute changes, go on stage with your buddies to present the prototype and boom! It's not working, because your last changes broke something important. If only you had tests to catch that error! Tests are great time-savers for the simple reason that they automate manual testing.
+
+How do you test applications in Node to avoid breaking existing functionality? You can use one of the most popular testing frameworks Mocha. This lesson will cover the basics of Mocha testing framework and Chai Expect behavioral-driven language to get you started with the labs.
+
+
+## Installing Mocha and Chai Expect
+
+First, we need to install Mocha and Chai. To do so, you'll need to open your terminal and create either the `node_modules` folder or the `package.json` file:
+
+```
+$ mkdir node_modules
+```
+
+```
+$ npm init
+```
+
+Then, you can use the `npm install NAME` to install Mocha and Chai Expect. We'll cover npm and modules later in more details.
+
+```
+$ npm install mocha@2.4.5 --save-dev
+$ npm install chai@3.5.0 --save-dev
+```
+
+That's it. In your `node_modules` you should see two new folders: `mocha` and `chai`, if you run `$ ls node_modules` to get the list of directories.
+
+Note: Chai comes with Expect so by installing Chai we are getting Chai Expect.
+
+## Writing Mocha Test
+
+Expect has a lot of features and if you're familiar with behavioral-driven development (like from Rspec and Jasmine), you can learn Expect fast. We won't be covering all of the features in the lesson, only the most important ones which you'll encounter in the labs of this course.
+
+You can create a file `test.js` to follow this exercise. So in the beginning, we need to import the module. 
+
+```js
+var expect = require('chai').expect
+```
+
+Then we need to create some objects which we will be testing. Typically these objects will come from other modules (unit testing):
+
+```js
+var name = 'React Quickly'
+var url = ['http://reactquickly.co', 'https://www.manning.com/books/react-quickly']
+```
+
+Next, we use `describe` with a string and function arguments to define the test suite:
+
+```js
+describe('name and url', function() {
+```
+
+The next method is `it()`, think about it as the test case:
+
+```js
+  it('must be a string', function(done){
+```
+
+We invoke `expect()` with the object which we want to test like `name` or `url`. Then, we use chained methods and properties like `to.be.a()` to check for the type (string) and value (React Quickly):
+
```

Comment:
add that expect actually calls your functions and tests your variables, uses a matcher like `to.be.a` and compares it to the value passed to the matcher